core/o11y/src/delay_detector.rs

```diff
@@ -0,0 +1,96 @@
+use crate::metrics::try_create_histogram_vec;
+use once_cell::sync::Lazy;
+use prometheus::{exponential_buckets, HistogramVec};
+use std::time::Instant;
+use tracing::span::Attributes;
+use tracing::Id;
+use tracing_subscriber::layer::Context;
+use tracing_subscriber::registry::LookupSpan;
+use tracing_subscriber::Layer;
+
+#[derive(Default)]
+pub(crate) struct DelayDetectorLayer {}
```

Comment:
How about TracingSpanDurationLogger or Subscriber?

Reply:
Thank you for the suggestion. `SpanDurationLogger` should be clear enough.